tests/dialects/test_postgres.py

```diff
@@ -577,13 +577,19 @@ def test_bool_or(self):
         )
 
     def test_string_concat(self):
+        self.validate_all(
+            "SELECT CONCAT('abcde', 2, NULL, 22)",
+            write={
+                "postgres": "SELECT CONCAT(COALESCE(CAST('abcde' AS TEXT), ''), COALESCE(CAST(2 AS TEXT), ''), COALESCE(CAST(NULL AS TEXT), ''), COALESCE(CAST(22 AS TEXT), ''))",
+            },
+        )
         self.validate_all(
             "CONCAT(a, b)",
             write={
-                "": "CONCAT(COALESCE(a, ''), COALESCE(b, ''))",
-                "duckdb": "CONCAT(COALESCE(a, ''), COALESCE(b, ''))",
-                "postgres": "CONCAT(COALESCE(a, ''), COALESCE(b, ''))",
-                "presto": "CONCAT(CAST(COALESCE(a, '') AS VARCHAR), CAST(COALESCE(b, '') AS VARCHAR))",
+                "": "CONCAT(COALESCE(CAST(a AS TEXT), ''), COALESCE(CAST(b AS TEXT), ''))",
+                "duckdb": "CONCAT(COALESCE(CAST(a AS TEXT), ''), COALESCE(CAST(b AS TEXT), ''))",
+                "postgres": "CONCAT(COALESCE(CAST(a AS TEXT), ''), COALESCE(CAST(b AS TEXT), ''))",
```

Comment:
so this is one of the downsides of this approach, postgres concat(a, b) would ideally stay as an identity, this is similar to tsordstodate, if we had created a SAFE_CONCAT, then we would get so much noise

Reply:
Discussed in Slack -- we already have a `SafeConcat` expression. The issue is that we have two axes of variability:

- The first one is “concat’s args must all be strings” --> controlled by the STRICT_STRING_CONCAT property
- The second one is “null values in a concat are interpreted as empty strings” --> controlled by the CONCAT_NULL_OUTPUTS_STRING property